applications/tari_console_wallet/src/grpc/wallet_grpc_server.rs

```diff
@@ -542,6 +551,41 @@ impl wallet_server::Wallet for WalletGrpcServer {
         Ok(Response::new(GetTransactionInfoResponse { transactions }))
     }
 
+    async fn stream_transaction_events(
+        &self,
+        _request: tonic::Request<TransactionEventRequest>,
+    ) -> Result<Response<Self::StreamTransactionEventsStream>, Status> {
+        let (mut sender, receiver) = mpsc::channel(100);
+        task::spawn(async move {
+            let event_listener = &WALLET_EVENT_LISTENER.receiver;
+            loop {
+                let transaction_event = match event_listener.recv_timeout(Duration::from_secs(1)) {
+                    Ok(source) => TransactionEvent::try_from(source),
+                    Err(_) => continue,
```

Comment:
This is essentially a spin-lock of sorts. Stanimal - is this the best practice approach here?

Reply:
You should be able to remove the timeout and use recv